src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -263,6 +269,51 @@ public BulkInsertOperation(string database, IDocumentStore store, BulkInsertOpti
 
             _generateEntityIdOnTheClient = new GenerateEntityIdOnTheClient(_requestExecutor.Conventions,
                 entity => AsyncHelpers.RunSync(() => _requestExecutor.Conventions.GenerateDocumentIdAsync(database, entity)));
+
+            _streamLock = new AsyncReaderWriterLock();
+            _lastWriteToStream = DateTime.Now;
+            _timer = new Timer( async o => await SendHeartBeat(o).ConfigureAwait(false),
+                null,
+                (int)(StreamWithTimeout.DefaultWriteTimeout.TotalMilliseconds / 3),
+                Timeout.InfiniteTimeSpan.Milliseconds);
+        }
+
+        internal async Task SendHeartBeat(object _)
+        {
+            if (DateTime.Now.Subtract(_lastWriteToStream).TotalSeconds > (StreamWithTimeout.DefaultWriteTimeout.TotalSeconds * 3 ))
```

Comment:
Add some explanation here (how much time we wait between heartbeats, when we stop sending the heartbeat)